aduro/model.py

```diff
@@ -0,0 +1,239 @@
+"""Data classes for Aduro"""
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Any, List
+
+from .helpers import try_convert_object
+
+TIMESTAMP_FORMAT = "%Y-%m-%dT%H:%M:%S.%fZ"
+
+# pylint: disable=unsupported-binary-operation
+
+
+@dataclass
+class Connection:
+    """Dataclass for device connection"""
+
+    timestamp: datetime
+    online: bool
+
+    @staticmethod
+    def from_dict(obj: Any) -> "Connection":
```

Comment:
This makes your intent clearer. 

```suggestion
    def from_dict(obj: dict) -> "Connection":
```

Reply:
And a curse on the house of Python for its lack of multiple constructors. :)